ccmbioinfo/lfs
# CARDS-1780: Configure/Implement data import from Clarity

This PR involves the clarity importer, and includes sample testing data

## To test (via Docker):
1. Build with `-Pdocker` and navigate to the compose-cluster folder.
2. Clean up any old instance of CARDS with `docker-compose down && docker-compose rm && docker volume prune -f && ./cleanup.sh`
3. Start up Docker with `python3 generate_compose_yaml.py --mssql --cards_project cards4prems --oak_filesystem --dev_docker_image --composum --adminer` (of note: `--adminer` to setup adminer, `--cards_project cards4prems` for the prems project, `--mssql` to setup the MS-SQL container, and `--custom_env_file mssql/mssql.env` to set it up with the environment variables necessary.)
  - As an aside, the `compose-cluster/mssql/mssql.env` file is there to show which environment variables are important to setup CARDs' clarity importer.
4. `docker-compose build && docker-compose up -d` to start up the server 
5. Navigate to http://localhost:1435/ to access adminer. Login with the following credentials:
- System: `MS SQL (beta)`
- Server: `mssql`
- Username: `sa`
- Password: `testPassword_`
- Database: `master`
6. On the left bar, click Import.
7. On the next screen, upload the file `cards/compose-cluster/mssql/importTestSql.sql` like so:
![image](https://user-images.githubusercontent.com/4656440/200427919-a8fc69a9-cf6e-494c-80d1-6d98ec9c1117.png)
- It should say "4 queries executed OK." if it was successful.
- You can add more rows if you want to test other scenarios
- I've also included a tool inside `compose-cluster/mssql/` called `generate_test_sql.py`. Call it with `python3 generate_test_sql.py test.sql -n 100` to generate 100 entries
9. Login to cards with the admin account
10. Navigate to `http://localhost:8080/Subjects.importClarity`
- You should receive a white page with `{"status":"success","message":"Data import started"}`
11. The patients and forms should be imported momentarily. I have noticed a delay of a minute or five when importing more than 1000 forms.
12. When finished, `docker-compose down && docker-compose rm && docker volume prune -f && ./cleanup.sh`
